api/src/pcapi/utils/decorators.py

```diff
@@ -0,0 +1,44 @@
+from functools import wraps
+from logging import Logger
+from typing import Any
+from typing import Callable
+from typing import Optional
+from typing import Type
+
+
+def retry(
+    exception: Type[Exception],
+    exception_handler: Optional[Callable] = None,
+    tries: int = 1,
+    logger: Logger | None = None,
+) -> Callable:
+    """
+    Retry func on a given exception n (or 1 by default) times.
+    Slightly adapt from https://gitlab.com/woob/woob/-/blob/master/woob/tools/decorators.py#L25
+    original from https://www.saltycrane.com/blog/2009/11/trying-out-retry-decorator-python/
+
+    Args:
+        exception (Exception): Which exception should be catched
+        exception_handler (Callable): Handler to encapsulate whatever logic you want in case exception is catched. Optionnal.
+        tries (int): How many times do you want to call `func`. Default to 1
```

Comment:
(détail / doc / nommage) J'ai toujours du mal avec les fonctions de ce genre. :) Est-ce que l'argument (ici `tries`) est le nombre **total** de tentatives, ou le nombre de **ré-essais** (sous-entendu, après une première tentative ayant débouché sur une erreur). Ici, **d'après l'implémentation**, c'est le nombre de ré-essais : si je passe `tries=1`, la fonction sera appelée **2 fois** au plus. Or la docstring dit autre chose : "how many times do you want to call `func` : [...] **1**".

Je vois plusieurs possibilités : 
- soit on modifie juste la documentation. Mais si on ne lit pas la doc et qu'on lit seulement l'appel au décorateur, on verra `@retry(tries=1)`, qui semble indiquer qu'on fait 1 seul tentative (alors que le décorateur en fera 2) ;
- soit on renomme l'argument en `max_attempts`, qui me semble plus explicitement indiquer : "je vais faire au plus `max_attempts` tentatives". (Évidemment, on devra légèrement changer la condition de sortie dans l'implémentation.)

Qu'en penses-tu ?

Reply:
Très juste, j’ai renommé en `max_attempts` qui est plus explicite et modifié la condition en conséquence @dbaty